core/src/main/scala/chisel3/Data.scala

```diff
@@ -657,13 +657,10 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
     requireIsHardware(this)
     topBindingOpt match {
       // DataView
-      case Some(ViewBinding(target)) => reify(target).ref
+      case Some(ViewBinding(target))           => reify(target).ref
       case Some(AggregateViewBinding(viewMap)) =>
-        viewMap.get(this) match {
-          case None => materializeWire() // FIXME FIRRTL doesn't have Aggregate Init expressions
-          // This should not be possible because Element does the lookup in .topBindingOpt
-          case x: Some[_] => throwException(s"Internal Error: In .ref for $this got '$topBindingOpt' and '$x'")
-        }
+        // TODO the wire isn't necessary when the view is an identity view, we could optimize in that case
+        materializeWire()
```

Comment:
"Take some pride in your work."

Reply:
FWIW I don't think that case is all that common in practice (mainly in tests) so I'm not sure it's really even worth it.

Reply:
My main thinking is that the effect of _any_ unnecessary wires, nodes, etc. are getting amplified by not having widespread auto-D/I or instantiate. Hence, the cost of any inefficient Chisel generation of FIRRTL goes through a multiplicative factor that can make innocuous wires suddenly measureable. CIRCT will clean up a lot of unneeded wires (@darthscsi has a pass that removes 50% of all wires in one design), yet that still requires going through a fat waist of Chisel memory usage, on-disk FIRRTL footprint, and CIRCT footprint.

If this is _definitely_ not that common, then it's way less of an issue. 😅

Reply:
Yeah unnecessary wires is definitely a thing to keep our eyes on so thanks for keeping me honest 😅. I really do think that this case will result in basically zero additional FIRRTL. I will go ahead and add the check though and can verify it by checking the CHIRRTL.

I'm curious for that 50% of wires being removed, what percentage of those are just us adding a Wire because FIRRTL doesn't have an expression for constructing aggregates... I'm sure a lot of it is unnecessary use of `WireInit` by users though. Maybe we can add a lint to firtool for wires that are a direct reassignment of the exact same type?

Reply:
I went ahead and did the check and you can see in the tests when wires are not created. Please re-review!